src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendRecorder.java

```diff
@@ -177,6 +180,16 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, final Build
 
         contrastSDK = VulnerabilityTrendHelper.createSDK(profile.getUsername(), profile.getServiceKey(), profile.getApiKey(), profile.getTeamServerUrl());
 
+        try {
+            final Organizations organizations = contrastSDK.getProfileDefaultOrganizations();
+            if (organizations == null || organizations.getOrganization() == null) {
+               throw new AbortException("Connection error: No organization found, Check your credentials and URL.");
+            }
+
+        } catch (UnauthorizedException | IOException e) {
+            throw new AbortException("Unable to connect to Contrast.");
```

Comment:
Lets add a log here to give more context to the error similar to how they are done later in the code 
`VulnerabilityTrendHelper.logMessage(listener, e.getMessage());`

Reply:
Lets also add the same to the pipeline-step.

Reply:
I was modeling it after the test connection in the config page which doesn't provide any context for error, but do you think we can because there's more room for a full stack trace?

Reply:
not sure. It's just the exception message. I opened a PR to add a message to the test connection https://github.com/jenkinsci/contrast-continuous-application-security-plugin/pull/70